lib/Control/Control.js

```diff
@@ -1663,7 +1663,7 @@ Control.concat = function (ctor, props, instance) {
 
 	if (props.style) {
 		if (instance) {
-			str = (proto.style ? (proto.style + ';') : '') + (props.style + ';');
+			str = (proto.style ? proto.style : '') + props.style;
```

Comment:
Hi @aarontam , I tested the solution with different combinations of parent kind's styles, its working fine in all the cases. I guess by design of 'control concat' methods, it doesn't create any issues. However  it is causing problems in `overrideComponets` block, I don't know exact reason, some analysis is required to identify the problem. But in either case your solution works, as other things will be taken care by `normalizeCssStyleString`